WooCommerce/Classes/ViewRelated/Orders/Order Creation/CustomAmounts/AddCustomAmountViewModel.swift

```diff
@@ -0,0 +1,56 @@
+import Combine
+import WooFoundation
+import UIKit
+import SwiftUI
+
+typealias CustomAmountEntered = (_ amount: String, _ name: String) -> Void
+
+final class AddCustomAmountViewModel: ObservableObject {
+    let formattableAmountTextFieldViewModel: FormattableAmountTextFieldViewModel
+    let onCustomAmountEntered: CustomAmountEntered
+
+    init(locale: Locale = Locale.autoupdatingCurrent,
+         storeCurrencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         onCustomAmountEntered: @escaping CustomAmountEntered) {
+        self.formattableAmountTextFieldViewModel = FormattableAmountTextFieldViewModel(locale: locale, storeCurrencySettings: storeCurrencySettings)
+        self.onCustomAmountEntered = onCustomAmountEntered
+        listenToAmountChanges()
+    }
+
+    /// Variable that holds the name of the custom amount.
+    ///
+    @Published var name = ""
+    @Published var shouldDisableDoneButton: Bool = true
+
+    var customAmountPlaceholder: String {
+        Localization.customAmountPlaceholder
+    }
+
+    func doneButtonPressed() {
+        let customAmountName = name.isNotEmpty ? name : customAmountPlaceholder
+        onCustomAmountEntered(formattableAmountTextFieldViewModel.amount, customAmountName)
+        reset()
+    }
+
+    func reset() {
+        name = ""
+        shouldDisableDoneButton = true
+
+        formattableAmountTextFieldViewModel.reset()
+    }
```

Comment:
Should we separate dismissal from reset? So it's clearer from the EditableViewModel when is triggered:
 ```suggestion
    func dismiss() {
      reset()
    }

    private func reset() {
        name = ""
        shouldDisableDoneButton = true
        formattableAmountTextFieldViewModel.reset()
    }
```

So when we call:
```diff
    func onDismissAddCustomAmountView() {
-        addCustomAmountViewModel.reset()
+         addCustomAmountViewModel.dismiss()
    }
```

Reply:
I think dismiss and reset are already split, they have their own functions and aren't coupled to each other. Calling `addCustomAmountViewModel.dismiss()` in `onDismissAddCustomAmountView` looks confusing to me, the view is already dismissed; and we don't want to dismiss it again.